okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -885,7 +899,17 @@ private void stopIfNecessary() {
 
     // We will close the underlying socket in the writing thread to break out the reader
     // thread, which will close the frameReader and notify the listener.
-    frameWriter.close();
+    // When the transport is middle of start and received stop before the frameWriter init
```

Comment:
That would be an invalid API call; start() must have returned before calling shutdown(). The only way it looks like that could happen is via KeepAliveManager (although it would be a _very_ rare race), because it is initialized before the lock (which could be fixed)